power_grid_model_c/power_grid_model/include/power_grid_model/common/exception.hpp

```diff
@@ -118,6 +118,13 @@ class InvalidRegulatedObject : public PowerGridError {
     }
 };
 
+class DuplicatedRegulatedObject : public PowerGridError {
+  public:
+    DuplicatedRegulatedObject() {
```

Comment:
`Duplicatively` then.

Reply:
Maybe even better tonswitch it around: `DuplicateRegulation` because it would be ok if multiple regulators regulate the same object, as long as they dont regulate the same attribute.

Similarly, there could be one single regulator regulating multiple attributes of a single object.

We can maybe also come up with a regulator that regulates multiple components at the same time. Maybe a smart switch that switches on and off certain loads, some of which have const power, others const current or const impedance, maybe some can be regulated, ... . Could be depending on e.g. line load or energy price.

Regulated object does not fit the purpose well then

Reply:
Or `ConflictingRegulation`

Reply:
`Conflicting...` isn't all that descriptive. `DuplicativelyRegulated...` means something is regulated by more than one party, which is perfect for our purpose. **If** there was some error for the regulator side, that it regulates more than it should, it would be `DuplicativelyRagulating...`. I don't think we are at the granularity of attributes. If that would be an error, it should be in consult with expert input. Otherwise, I wouldn't create imaginary cases (yet).